### PR TITLE
feat: configures monitoring pod resources for Arc extension

### PR DESCRIPTION
**Pull request by nilekhc**
**Reason for Change**:

**Requirements**

- [x] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: manifest_staging/charts/csi-secrets-store-provider-azure/README.md [hunk 1]
```diff
@@ -154,3 +154,9 @@ The following table lists the configurable parameters of the csi-secrets-store-p
...
 | `writeCertAndKeyInSeparateFiles`                                 | Write cert and key in separate files. The individual files will be named as <secret-name>.crt and <secret-name>.key. These files will be created in addition to the single file.                      | `false`                                                                                          |
 | `metricsAddr`                                                    | Port that serves metrics                                                                                                                                                                              | `8898`                                                                                           |
+| `promMdmConverter.resources`                                     | Resource limit for Arc ext monitoring pod for promMdmConverter container                                                                                                                              | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/README.md:157]:
```suggestion
| `promMdmConverter.resources`                                     | Resource limit for Arc ext monitoring pod's prom-mdm-converter container                                                                                                                              | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
```

File: manifest_staging/charts/csi-secrets-store-provider-azure/README.md [hunk 2]
```diff
@@ -154,3 +154,9 @@ The following table lists the configurable parameters of the csi-secrets-store-p
...
+| `mdm.resources`                                                  | Resource limit for Arc ext monitoring pod for mdm container                                                                                                                                           | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/README.md:158]:
```suggestion
| `mdm.resources`                                                  | Resource limit for Arc ext monitoring pod's mdm container                                                                                                                                           | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
```

File: manifest_staging/charts/csi-secrets-store-provider-azure/README.md [hunk 3]
```diff
@@ -154,3 +154,9 @@ The following table lists the configurable parameters of the csi-secrets-store-p
...
+| `msiAdapter.resources`                                           | Resource limit for Arc ext monitoring pod for msiAdapter container                                                                                                                                    | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/README.md:159]:
```suggestion
| `msiAdapter.resources`                                           | Resource limit for Arc ext monitoring pod's msi-adapter container                                                                                                                                    | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
```

File: manifest_staging/charts/csi-secrets-store-provider-azure/README.md [hunk 4]
```diff
@@ -154,3 +154,9 @@ The following table lists the configurable parameters of the csi-secrets-store-p
...
+| `telegraf.resources`                                             | Resource limit for Arc ext monitoring pod for telegraf container                                                                                                                                      | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/README.md:160]:
```suggestion
| `telegraf.resources`                                             | Resource limit for Arc ext monitoring pod for telegraf container                                                                                                                                      | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
```
```suggestion
| `telegraf.resources`                                             | Resource limit for Arc ext monitoring pod's telegraf container                                                                                                                                      | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
```

File: manifest_staging/charts/csi-secrets-store-provider-azure/README.md [hunk 5]
```diff
@@ -154,3 +154,9 @@ The following table lists the configurable parameters of the csi-secrets-store-p
...
+| `amacoreagent.resources`                                         | Resource limit for Arc ext monitoring pod for amacoreagent container                                                                                                                                  | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/README.md:161]:
```suggestion
| `amacoreagent.resources`                                         | Resource limit for Arc ext monitoring pod's amacoreagent container                                                                                                                                  | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
```

File: manifest_staging/charts/csi-secrets-store-provider-azure/README.md [hunk 6]
```diff
@@ -154,3 +154,9 @@ The following table lists the configurable parameters of the csi-secrets-store-p
...
+| `fluentd.resources`                                              | Resource limit for Arc ext monitoring pod for fluentd container                                                                                                                                       | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/README.md:162]:
```suggestion
| `fluentd.resources`                                              | Resource limit for Arc ext monitoring pod's fluentd container                                                                                                                                       | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
```

File: manifest_staging/charts/csi-secrets-store-provider-azure/arc-values.yaml [hunk 7]
```diff
@@ -171,6 +171,54 @@ constructPEMChain: true
...
+      cpu: 50m
+      memory: 100Mi
+fluentd:
+  resources:
+    requests:
+      cpu: 50m
+      memory: 100Mi
+    limits:
+      cpu: 50m
+      memory: 100Mi
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/arc-values.yaml:221]: For `fluentd`, the current hardcoded limits are

https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/ecaccb15f119e8c2cd47be70cf602e1f38eb307a/charts/csi-secrets-store-provider-azure/templates/arc-monitoring.yaml#L184-L193

Is `100Mi` sufficient?

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/arc-monitoring.yaml [hunk 8]
```diff
@@ -101,12 +86,7 @@ spec:
           image: "mcr.microsoft.com/oss/mirror/docker.io/library/telegraf:1.21"
           imagePullPolicy: IfNotPresent
           resources:
-            requests:
-              cpu: 50m
-              memory: 100Mi
-            limits:
-              cpu: 50m
-              memory: 100Mi
+            {{- toYaml .Values.msiAdapter.resources | nindent 12 }}
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/templates/arc-monitoring.yaml:89]:
```suggestion
            {{- toYaml .Values.telegraf.resources | nindent 12 }}
```

**Review by aramase — CHANGES_REQUESTED** [review]

File: manifest_staging/charts/csi-secrets-store-provider-azure/arc-values.yaml (hunk 7 above)
Comment (nilekhc) [reply manifest_staging/charts/csi-secrets-store-provider-azure/arc-values.yaml:221]: This is the default limit we decided to set based on our provider pod defaults. It's works for most users. Making it configurable would help with edge cases scenarios where user need to increase them due to higher consumption.

Comment (aramase) [reply manifest_staging/charts/csi-secrets-store-provider-azure/arc-values.yaml:221]: In this PR we're changing to default to `100Mi` from the previous `250Mi`. Is that intended?

Comment (nilekhc) [reply manifest_staging/charts/csi-secrets-store-provider-azure/arc-values.yaml]: reverted to original defaults.

File: manifest_staging/charts/csi-secrets-store-provider-azure/README.md [hunk 9]
```diff
@@ -154,3 +154,9 @@ The following table lists the configurable parameters of the csi-secrets-store-p
...
+| `telegraf.resources`                                             | Resource limit for Arc ext monitoring pod's telegraf container                                                                                                                                        | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
+| `amacoreagent.resources`                                         | Resource limit for Arc ext monitoring pod's amacoreagent container                                                                                                                                    | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
+| `fluentd.resources`                                              | Resource limit for Arc ext monitoring pod's fluentd container                                                                                                                                         | `requests.cpu: 50m`<br>`requests.memory: 100Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 100Mi`   |
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/README.md:162]:
```suggestion
| `fluentd.resources`                                              | Resource limit for Arc ext monitoring pod's fluentd container                                                                                                                                         | `requests.cpu: 50m`<br>`requests.memory: 250Mi`<br>`limits.cpu: 50m`<br>`limits.memory: 250Mi`   |
```

**Review by aramase — APPROVED** [review]

**Pull request merged**